service/storage_service.cc

```diff
@@ -564,7 +564,8 @@ future<storage_service::nodes_to_notify_after_sync> storage_service::sync_raft_t
             tmptr->add_leaving_endpoint(host_id);
             co_await update_topology_change_info(tmptr, ::format("{} {}/{}", rs.state, id, ip));
```

Comment:
> ```
> At this point a replacing node will already make sure that its
> topology state is up-to-date and it knows coordinator's topology.
> ```
> 
> Why are we sure of that? 

Because it will execute a barrier in respond_to_joining_node rpc handler. The one that currently fails.